### PR TITLE
Version/1.12

### PR DESCRIPTION
**Pull request by bob0bob**
Closes #
Closes #
Closes #

# Changes proposed in this pull request:
-
-
-

Review please

### CODE REVIEW

**Comment by bob0bob** [comment]
please close.  mistake.  trying to pull into my branch for working.

File: src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/PlayerRequestResolver.java [hunk 1]
```diff
@@ -121,7 +119,7 @@ public IRequest getFollowupRequestForCompletion(@NotNull final IRequestManager m
     @Override
     public IRequest onRequestCancelledOrOverruled(@NotNull final IRequestManager manager, @NotNull final IRequest request) throws IllegalArgumentException
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/PlayerRequestResolver.java:120]: MINOR Remove the declaration of thrown exception 'java.lang.IllegalArgumentException' which is a runtime exception. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java [hunk 2]
```diff
@@ -132,20 +129,8 @@ public EntityAIWorkDeliveryman(@NotNull final JobDeliveryman deliveryman)
      */
     public AIState gather()
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java:130]: MAJOR Reduce the number of returns of this method 7, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java [hunk 3]
```diff
@@ -68,22 +70,17 @@
...
-     * Next target the deliveryman should gather stuff at.
-     */
-    private BlockPos gatherTarget = null;
-
-    /**
-     * Tracks how many buildings the DMan visited to gather resources.
+     * Warehouse the deliveryman is assigned to.
      */
-    private int gatherCount = 0;
+    private BuildingWareHouse wareHouse = null;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java:78]: MAJOR Remove this unused "wareHouse" private field. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1068)

File: src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/BuildingRequestResolver.java [hunk 4]
```diff
@@ -96,7 +105,16 @@ public boolean canResolve(
     public void resolve(
                          @NotNull final IRequestManager manager, @NotNull final IRequest<? extends IDeliverable> request) throws RuntimeException
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/BuildingRequestResolver.java:106]: MINOR Remove the declaration of thrown exception 'java.lang.RuntimeException' which is a runtime exception. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

File: src/main/java/com/minecolonies/coremod/colony/requestsystem/requesters/BuildingBasedRequester.java [hunk 5]
```diff
@@ -9,12 +9,10 @@
 import com.minecolonies.coremod.colony.ColonyManager;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.util.text.ITextComponent;
-import net.minecraft.util.text.TextComponentString;
 import net.minecraft.world.World;
+import net.minecraftforge.fml.common.FMLCommonHandler;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/colony/requestsystem/requesters/BuildingBasedRequester.java:13]: MINOR Remove this unused import 'net.minecraftforge.fml.common.FMLCommonHandler'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 125 issues
* ![CRITICAL][CRITICAL] 7 critical
* ![MAJOR][MAJOR] 34 major
* ![MINOR][MINOR] 79 minor
* ![INFO][INFO] 5 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [WindowTownHall.java#L351](https://github.com/Minecolonies/minecolonies/blob/72933c0720619da7069583d63c0f5654d7ba5b8d/src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java#L351): Use already-defined constant 'BUTTON_ACTIONS' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [WindowTownHall.java#L698](https://github.com/Minecolonies/minecolonies/blob/72933c0720619da7069583d63c0f5654d7ba5b8d/src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java#L698): Use already-defined constant 'BUTTON_TRIGGER' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [RequestHandler.java#L88](https://github.com/Minecolonies/minecolonies/blob/72933c0720619da7069583d63c0f5654d7ba5b8d/src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/RequestHandler.java#L88): Add a default case to this switch. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)
1. ![CRITICAL][CRITICAL] [AbstractRequest.java#L450](https://github.com/Minecolonies/minecolonies/blob/72933c0720619da7069583d63c0f5654d7ba5b8d/src/main/java/com/minecolonies/coremod/colony/requestsystem/requests/AbstractRequest.java#L450): Missing curly brace. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00121)
1. ![CRITICAL][CRITICAL] [BuildingRequestResolver.java#L151](https://github.com/Minecolonies/minecolonies/blob/72933c0720619da7069583d63c0f5654d7ba5b8d/src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/BuildingRequestResolver.java#L151): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1186)
1. ![CRITICAL][CRITICAL] [BuildingRequestResolver.java#L158](https://github.com/Minecolonies/minecolonies/blob/72933c0720619da7069583d63c0f5654d7ba5b8d/src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/BuildingRequestResolver.java#L158): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1186)
1. ![CRITICAL][CRITICAL] [PlayerRequestResolver.java#L155](https://github.com/Minecolonies/minecolonies/blob/72933c0720619da7069583d63c0f5654d7ba5b8d/src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/PlayerRequestResolver.java#L155): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1186)
1. ![MAJOR][MAJOR] [BuildingLumberjack.java#L277](https://github.com/Minecolonies/minecolonies/blob/72933c0720619da7069583d63c0f5654d7ba5b8d/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingLumberjack.java#L277): Reduce this class from 70 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [RequestHandler.java#L29](https://github.com/Minecolonies/minecolonies/blob/72933c0720619da7069583d63c0f5654d7ba5b8d/src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/RequestHandler.java#L29): Add a private constructor to hide the implicit public one. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1118)
1. ![MAJOR][MAJOR] [RequestHandler.java#L114](https://github.com/Minecolonies/minecolonies/blob/72933c0720619da7069583d63c0f5654d7ba5b8d/src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/RequestHandler.java#L114): Refactor this method to reduce its Cognitive Complexity from 36 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request closed without merging**